### PR TITLE
Add python-blosc2 recipe

### PR DESCRIPTION
**Pull request by goanpeca**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- ~[ ] If static libraries are linked in, the license of the static library is packaged.~
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/blosc2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ base_name[0] }}/{{ base_name }}/{{ base_name }}-{{ version }}.tar.gz
+  sha256: 63606498aaa72d58215b618d4512d5d3de29000a7b01a870edce8cb21d237c40
+  patches:
+    - 0001-Add-imported-target-with-pkg-config-to-support-windo.patch
```
Comment (isuruf) [inline recipes/blosc2/meta.yaml:13]:
```suggestion
    # Delete after https://github.com/Blosc/python-blosc2/pull/148 is merged
    - 0001-Add-imported-target-with-pkg-config-to-support-windo.patch
```

File: recipes/blosc2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - python                              # [build_platform != target_platform]
+  host:
+    - c-blosc2
+    - cython >=3
+    - cmake
+    - make
+    - ninja
+    - numpy >=1.20.3
+    - pathlib
+    - pkg-config
```
Comment (isuruf) [inline recipes/blosc2/meta.yaml:35]: Move to `build`

File: recipes/blosc2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    - {{ compiler("cxx") }}
+    - c-blosc2
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - python                              # [build_platform != target_platform]
+  host:
+    - c-blosc2
+    - cython >=3
+    - cmake
+    - make
+    - ninja
```
Comment (isuruf) [inline recipes/blosc2/meta.yaml:32]: These three too

File: recipes/blosc2/meta.yaml (hunk 2 above)
Comment (goanpeca) [reply recipes/blosc2/meta.yaml]: Fixed!

File: recipes/blosc2/meta.yaml (hunk 3 above)
Comment (goanpeca) [reply recipes/blosc2/meta.yaml]: Fixed

**Comment by FrancescAlted** [comment]
> @FrancescAlted, @martaiborra, do you want to be added as maintainers of the recipe?

No problem with me.  I don't have a lot of bandwidth (as anyone else, I suppose :-), but always happy to contribute if time allows.

**Comment by martaiborra** [comment]
> @FrancescAlted, @martaiborra, do you want to be added as maintainers of the recipe?

Same for me.

**Comment by goanpeca** [comment]
This one is ready for review @conda-forge/staged-recipes :)

**Comment by goanpeca** [comment]
>I don't master cmake enough to judge the proposed patch for CMakeLists.txt file, but other than that, this looks good to me. A minor nit: why 2.2.9 when 2.3.1 is out? [Blosc/python-blosc2/releases](https://github.com/Blosc/python-blosc2/releases)

It was the version that was available when I made the pull request, it can be updated as soon as this is merged :)

**Pull request merged**